function/main.py

```diff
@@ -0,0 +1,159 @@
+
+    try:
+        job_id = request_json['jobId']
+    except:
+        return handle_exception("error", "Could not find a jobId in the following request made at " + start_time + ": " + str(request.json))
+
+    write_running_to_bucket(job_id, start_time)
+
+    try:
+        url = request_json['url']
+        user = request_json['user']
+        user_email = user['userEmail']
+        user_subject_id = user['userSubjectId']
+        workspace = request_json['workspace']
+        workspace_name = workspace['name']
+        workspace_namespace = workspace['namespace']
+    except KeyError as ke:
+        return handle_exception(job_id, "Key Error: " + str(ke) + " in request " + str(request.json))
+
+    defaults = {'b64-decode-enums': True, 'prefix-object-ids': True}
+    request_options = request_json.get('options', {})
+    options = {**defaults, **request_options}
+
+    try:
+        avro = urllib.request.urlopen(url)
+    except urllib.error.URLError as ue:
+        return handle_exception(job_id, start_time, "URL Error - the url " + url + " is not valid.")
+
+    try:
+        reader = fastavro.reader(avro)
+        translation = Translator(options).translate(reader)
+        metadata = {
+            "namespace": workspace_namespace,
+            "name": workspace_name,
+            "userSubjectId": user_subject_id,
+            "userEmail": user_email,
+            "jobId": job_id,
+            "startTime": start_time,
+        }
+        metadata_json_str = json.dumps(metadata)
+        metadat_file_name = job_id + "/metadata.json"
+        upsert_json_str = json.dumps(translation)
+        upsert_file_name = job_id + "/upsert.json"
+        write_to_bucket(metadat_file_name, metadata_json_str)
+        write_to_bucket(upsert_file_name, upsert_json_str)
+    except Exception as e:
+        return handle_exception(job_id, "The following exception occurred: " + str(e))
+
+
+def handle_exception(subdirectory_name, message):
+    error_time = str(datetime.now())
+    error_file_name = subdirectory_name + "/error.json"
+    contents = json.dumps({ "error_time": error_time, "error_message": message})
+    write_to_bucket(error_file_name, contents)
+    return message
+
+
+def write_upsert_to_bucket(job_id, content_string):
+    compressed_value = gzip.compress(bytes(content_string, 'utf-8'))
+    file_name = job_id + "/upsert.json"
+    write_to_bucket(file_name, compressed_value)
+
+
+def write_metadata_to_bucket(job_id, content_string):
+    file_name = job_id + "/metadata.json"
+    write_to_bucket(file_name, content_string)
+
+
+def write_running_to_bucket(job_id, start_time):
+    file_name = job_id + "/running.json"
+    contents = json.dumps({"start_time": start_time})
+    write_to_bucket(file_name, contents)
+
+
+def write_to_bucket(file_name, content):
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket("avro-translated-json")
```

Comment:
bucket name needs to come from config to get the `-$ENV` suffix

Reply:
I added the ENV as an environment variable for the cloud function and am reading it from there